programs/autocrat_v0/src/lib.rs

```diff
@@ -151,147 +152,84 @@ pub mod autocrat_v0 {
+            pass_market.quote_mint == quote_vault.conditional_on_finalize_token_mint,
             AutocratError::InvalidMarket
         );
         require!(
-            openbook_fail_market.quote_lot_size == 100,
+            fail_market.base_mint == base_vault.conditional_on_revert_token_mint,
             AutocratError::InvalidMarket
         );
         require!(
-            openbook_fail_market.collect_fee_admin == dao.treasury,
+            fail_market.quote_mint == quote_vault.conditional_on_revert_token_mint,
             AutocratError::InvalidMarket
         );
-        let clock = Clock::get()?;
 
-        let openbook_twap_pass_market = &ctx.accounts.openbook_twap_pass_market;
-        let openbook_twap_fail_market = &ctx.accounts.openbook_twap_fail_market;
-
-        require!(
-            openbook_twap_pass_market.twap_oracle.initial_slot + 50 >= clock.slot,
-            AutocratError::TWAPMarketTooOld
-        );
-        require!(
-            openbook_twap_fail_market.twap_oracle.initial_slot + 50 >= clock.slot,
-            AutocratError::TWAPMarketTooOld
-        );
-        require_eq!(
-            openbook_twap_pass_market
-                .twap_oracle
-                .max_observation_change_per_update_lots,
-            dao.max_observation_change_per_update_lots,
-            AutocratError::TWAPOracleWrongChangeLots
-        );
-        require_eq!(
-            openbook_twap_fail_market
-                .twap_oracle
-                .max_observation_change_per_update_lots,
-            dao.max_observation_change_per_update_lots,
-            AutocratError::TWAPOracleWrongChangeLots
-        );
-        require!(
-            openbook_twap_pass_market.twap_oracle.expected_value == dao.twap_expected_value,
-            AutocratError::TWAPMarketInvalidExpectedValue
-        );
-        require!(
-            openbook_twap_fail_market.twap_oracle.expected_value == dao.twap_expected_value,
-            AutocratError::TWAPMarketInvalidExpectedValue
-        );
-
-        let proposal = &mut ctx.accounts.proposal;
-
-        proposal.number = dao.proposal_count;
-        dao.proposal_count += 1;
-
-        // least signficant 32 bits of nonce are proposal number
-        // most significant bit of nonce is 0 for base (META) and 1 for quote (USDC)
-        require!(
-            ctx.accounts.base_vault.nonce == proposal.number as u64,
-            AutocratError::InvalidVaultNonce
-        );
-        require!(
-            ctx.accounts.quote_vault.nonce == (proposal.number as u64 | (1 << 63)),
-            AutocratError::InvalidVaultNonce
-        );
+        for market in [&pass_market, &fail_market] {
+            // The market expires a minimum of 7 days after the end of a 3 day proposal.
+            // Make sure to do final TWAP crank after the proposal period has ended
+            // and before the market expires, or else! Allows for rent retrieval from openbook
+            require!(
```

Comment:
It's more of an optional small thing but this might be an opportunity to switch from `require` to `require_gt` or `require_eq`. Just because I think that these other require macros will print out the values on both sides if anything goes wrong, rather than just print require evaluated to false + the error. Very minor thing and not worth risking messing up the existing checks over, so just mentioning it.

Reply:
yep, good catch! fixing now